src/monodroid/jni/monodroid-glue.c

```diff
@@ -686,13 +710,22 @@ get_libmonosgen_path ()
 	TRY_LIBMONOSGEN (get_libmonoandroid_directory_path ())
 #endif
 
-	TRY_LIBMONOSGEN (SYSTEM_LIB_PATH)
-	
-#ifdef RELEASE
-	log_fatal (LOG_DEFAULT, "cannot find libmonosgen-2.0.so in app_libdir: %s nor in previously printed locations.", app_libdir);
-#else
-	log_fatal (LOG_DEFAULT, "cannot find libmonosgen-2.0.so in override_dir: %s, app_libdir: %s nor in previously printed locations.", override_dirs[0], app_libdir);
+	TRY_LIBMONOSGEN (SYSTEM_LIB_PATH);
+	log_fatal (LOG_DEFAULT, "Cannot find '%s'. Looked in the following locations:", MONO_SGEN_SO);
+
+#ifndef RELEASE
+	if (!embedded_dso_mode) {
```

Comment:
This condition concerns me, in that it means that if a *Debug* build was using "embedded DSO support," that Debug build can't use Improved Fast Deployment, because we only look in the override directories when `embedded_dso_mode`==0.